data_prep/reuters_graph_datasets.py

```diff
@@ -0,0 +1,208 @@
+        train_mask[val_num:val_num + train_num] = 1
+
+        # Mask all non-test docs
+        test_mask = torch.zeros(len(self.iton), device=self.device)
+        test_mask[val_num + train_num:val_num + train_num + test_num] = 1
+
+        return train_mask.bool(), val_mask.bool(), test_mask.bool()
+
+    @staticmethod
+    def prepare_reuters(r8=False, val_size=0.1):
+        """
+        Filters out all documents which have more or less than 1 class.
+        Then filters out all classes which have no remaining documents.
+        Args:
+            r8 (bool, optional): R8 is constructed by taking only the top 10 (original) classes. Defaults to False.
+            val_size (float, optional): Proportion of training documents to include in the validation set.
+        Returns:
+            doc_splits (tuple): Tuple containing 3 List of training, test, and validation documents.
+            unique_classes (List): List of Strings containing the class names sorted in alphabetical order.
+        """
+        # Filter out docs which don't have exactly 1 class
+        data = defaultdict(lambda: {'train': [], 'test': []})
+
+        for doc in reuters.fileids():
+            categories = reuters.categories(doc)
+            if len(categories) == 1:
+                if doc.startswith('training'):
+                    data[categories[0]]['train'].append(doc)
+                if doc.startswith('test'):
+                    data[categories[0]]['test'].append(doc)
+
+        # Filter out classes which have no remaining docs
+        for cat in reuters.categories():
+            if len(data[cat]['train']) < 1 or len(data[cat]['test']) < 1:
+                data.pop(cat, None)
+
+        if r8:
+            # Choose top 10 classes and then select the ones which still remain after filtering
+            popular = sorted(reuters.categories(), key=lambda clz: len(reuters.fileids(clz)), reverse=True)[:10]
+            data = dict([(cls, splits) for (cls, splits) in data.items() if cls in popular])
+
+        # Create splits
+        train_docs = [doc for cls, splits in data.items() for doc in splits['train']]
+        test_docs = [doc for cls, splits in data.items() for doc in splits['test']]
+
+        # Select the validation documents out of the training documents
+        val_size = int(len(train_docs) * val_size)
+        random.shuffle(train_docs)
+        val_docs = train_docs[:val_size]
+        train_docs = train_docs[val_size:]
+
+        # sort the unique classes to ensure constant order
+        unique_classes = sorted(data.keys())
+
+        # For testing with only a few docs:
+        # return (train_docs[:200], test_docs[:20], val_docs[:20]), unique_classes
+
+        return (train_docs, test_docs, val_docs), unique_classes
+    
+    def len(self):
+        length = len(self.data)
+        return 1
+    
+    def get(self, idx):
+        return self.data
+
+
+class R52(Reuters):
```

Comment:
This will have to be changed to R52Graph

Reply:
Okay, I copied this from the other Reuters_Graph python file. Did you make any adaptions to R52 there already that are now missing here? In that case, we can just add them here as well

Reply:
Done